### ISSUE TITLE
Beatmap volume is decreased after epilepsy warning

### ISSUE THREAD

**Issue by Sinamer** [ppy/osu#10567]
**Describe the bug:** The volume on epilepsy warning screen gets lower, but doesn't return to normal when beatmap starts. The difference is easily noticeable after you retry the map, when the warning doesn't pop up.

**Screenshots or videos showing encountered issue:** https://streamable.com/aym55g

**osu!lazer version:** 2020.1020.0

**Logs:**

[runtime.log](https://github.com/ppy/osu/files/5409536/runtime.log)

**Comment by bdach** [comment]
Yeah, the adjustment is being removed on disposal which is not correct...

**Comment by bdach** [comment]
I've been looking at this for the past ~hour or so and the best I can do is [this](https://github.com/ppy/osu/compare/master...bdach:epilepsy-warning-volume-restoration) branch.

tl;dr:

* Old code did not wait for the fade of the warning to complete
* Therefore, the expire would only take effect 500ms after being called...
* which never happened, as another screen was pushed, blocking transform updates and ensuring the drawable is not expired, removed, and disposed (in that order).

I tried fudging the transform sequence to keep the expire there but it has to be on screen exit to avoid the volume going back up before push (and it's also finicky as you have to explicitly delay until the transform has definitely completed). I've also simplified the transform sequence as three nested schedules were giving me a headache - not sure whether there was a reason for structuring them like they are right now...

**Comment by smoogipoo** [comment]
Can the adjustment be added + removed during PopIn/PopOut?

**Comment by peppy** [comment]
That's how it was, but it brought the volume up before going down again and felt really bad.

Might be easier to move the adjustment out of the epilepsy warning, into `PlayerLoader` itself

**Comment by bdach** [comment]
Wouldn't entirely be a fan of the logic being spread around two places... Maybe could go for the inverse of what I had, by having a DI-able `PlayerLoader`, which would expose callbacks on close, to which the warnings would bind to in order to remove the adjustment?
